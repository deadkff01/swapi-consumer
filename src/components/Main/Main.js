import React, { useState, useEffect } from 'react'
import baseService from '../../services/baseService'
import Row from '../Common/Row'
import MoviesList from '../Common/MoviesList'
import Loader from '../Common/Loader'
import ErrorWarning from '../Common/ErrorWarning'

/**
 * Main component
 */
const Main = () => {
  const [isLoading, setIsloading] = useState(true)
  const [movies, setMovies] = useState([])
  const [requestError, setRequestError] = useState(false)

  const getMovies = async () => {
    try {
      const response = await baseService().get('films')
      const { results } = response.data
      setMovies(results)
      setIsloading(false)
    } catch {
      setIsloading(false)
      setRequestError(true)
    }
  }

  useEffect(() => {
    getMovies()
  }, [])

  if (requestError) {
    return (
      <ErrorWarning data-testid="component-error">
        Error, verify your conection or reload the page! may the force be with you!
      </ErrorWarning>
    )
  }

  return (
    <div data-testid="component-main">
      {isLoading ? (
        <Loader data-testid="loading" />
      ) : (
        <Row data-testid="resolved">
          <MoviesList movies={movies} />
        </Row>
      )}
    </div>
  )
}

export default Main
