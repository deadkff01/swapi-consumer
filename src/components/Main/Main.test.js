import React from 'react'
import Main from './Main'
import { render } from '@testing-library/react'

it('renders the component', () => {
  const container = render(<Main />)
  expect(container).toMatchSnapshot()
})
