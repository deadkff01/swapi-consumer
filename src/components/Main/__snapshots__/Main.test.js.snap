// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  class="sc-bdVaJa crJopo"
  data-testid="resolved"
>
  <div
    class="sc-bwzfXH KnFUT"
  >
    <div
      alt="Card of The Phantom Menace"
      class="sc-htpNat ijQqde"
    >
      <div
        alt="The Phantom Menace"
        class="sc-bxivhb evnSWs starwarts-movie-image-1"
        title="The Phantom Menace"
      />
      <div
        class="sc-ifAKCX gclsdh"
      >
        <h1>
          The Phantom Menace
        </h1>
        <a
          href="/movie-info"
        >
          <button
            class="sc-EHOje dVuMfl"
          >
            MORE INFO
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    class="sc-bwzfXH KnFUT"
  >
    <div
      alt="Card of Attack of the Clones"
      class="sc-htpNat ijQqde"
    >
      <div
        alt="Attack of the Clones"
        class="sc-bxivhb evnSWs starwarts-movie-image-2"
        title="Attack of the Clones"
      />
      <div
        class="sc-ifAKCX gclsdh"
      >
        <h1>
          Attack of the Clones
        </h1>
        <a
          href="/movie-info"
        >
          <button
            class="sc-EHOje dVuMfl"
          >
            MORE INFO
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    class="sc-bwzfXH KnFUT"
  >
    <div
      alt="Card of Revenge of the Sith"
      class="sc-htpNat ijQqde"
    >
      <div
        alt="Revenge of the Sith"
        class="sc-bxivhb evnSWs starwarts-movie-image-3"
        title="Revenge of the Sith"
      />
      <div
        class="sc-ifAKCX gclsdh"
      >
        <h1>
          Revenge of the Sith
        </h1>
        <a
          href="/movie-info"
        >
          <button
            class="sc-EHOje dVuMfl"
          >
            MORE INFO
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    class="sc-bwzfXH KnFUT"
  >
    <div
      alt="Card of A New Hope"
      class="sc-htpNat ijQqde"
    >
      <div
        alt="A New Hope"
        class="sc-bxivhb evnSWs starwarts-movie-image-4"
        title="A New Hope"
      />
      <div
        class="sc-ifAKCX gclsdh"
      >
        <h1>
          A New Hope
        </h1>
        <a
          href="/movie-info"
        >
          <button
            class="sc-EHOje dVuMfl"
          >
            MORE INFO
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    class="sc-bwzfXH KnFUT"
  >
    <div
      alt="Card of The Empire Strikes Back"
      class="sc-htpNat ijQqde"
    >
      <div
        alt="The Empire Strikes Back"
        class="sc-bxivhb evnSWs starwarts-movie-image-5"
        title="The Empire Strikes Back"
      />
      <div
        class="sc-ifAKCX gclsdh"
      >
        <h1>
          The Empire Strikes Back
        </h1>
        <a
          href="/movie-info"
        >
          <button
            class="sc-EHOje dVuMfl"
          >
            MORE INFO
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    class="sc-bwzfXH KnFUT"
  >
    <div
      alt="Card of Return of the Jedi"
      class="sc-htpNat ijQqde"
    >
      <div
        alt="Return of the Jedi"
        class="sc-bxivhb evnSWs starwarts-movie-image-6"
        title="Return of the Jedi"
      />
      <div
        class="sc-ifAKCX gclsdh"
      >
        <h1>
          Return of the Jedi
        </h1>
        <a
          href="/movie-info"
        >
          <button
            class="sc-EHOje dVuMfl"
          >
            MORE INFO
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    class="sc-bwzfXH KnFUT"
  >
    <div
      alt="Card of The Force Awakens"
      class="sc-htpNat ijQqde"
    >
      <div
        alt="The Force Awakens"
        class="sc-bxivhb evnSWs starwarts-movie-image-7"
        title="The Force Awakens"
      />
      <div
        class="sc-ifAKCX gclsdh"
      >
        <h1>
          The Force Awakens
        </h1>
        <a
          href="/movie-info"
        >
          <button
            class="sc-EHOje dVuMfl"
          >
            MORE INFO
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
